### ISSUE TITLE
Custom component development

### ISSUE THREAD

**Issue by diggit** [pyocd/pyOCD#1366]
Hi,
Let's say I want to add support for another RTOS, but I don't want to edit pyOCD sources directly. Is pyOCD able to load external components?
e.g. pemicro probe is standalone python package, so I guess it should be possible...

**Comment by flit** [comment]
Yup! To some extent, at least. Pyocd currently supports RTOS and debug probe plugins. So it sounds like you should be ok.

(I'm converting this issue to a discussion and will continue with more details there.)
